contrib/format-maprdb/src/main/java/org/apache/drill/exec/store/mapr/db/json/MaprDBJsonRecordReader.java

```diff
@@ -448,7 +448,11 @@ public int next() {
     }
 
     if (nonExistentColumnsProjection && recordCount > 0) {
-      JsonReaderUtils.ensureAtLeastOneField(vectorWriter, getColumns(), schema, allTextMode, Collections.emptyList());
+      if (schema != null) {
+        JsonReaderUtils.writeColumnsFromSchema(vectorWriter, getColumns(), schema, allTextMode);
```

Comment:
Schema can be without columns, this way we should fall back to `JsonReaderUtils.ensureAtLeastOneField`.